common/src/main/java/tachyon/heartbeat/HeartbeatScheduler.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Licensed to the University of California, Berkeley under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.heartbeat;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.locks.Condition;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
+/**
+ * This class can be used for controlling heartbeat execution.
+ */
+public final class HeartbeatScheduler {
+  private static Map<String, ScheduledTimer> sTimers = new HashMap<String, ScheduledTimer>();
+  private static Lock sLock =  new ReentrantLock();
+  private static Condition sCondition = sLock.newCondition();
+
+  /**
+   * @param timer a timer to add to the scheduler
+   */
+  public static void addTimer(ScheduledTimer timer) {
+    sLock.lock();
+    sTimers.put(timer.getThreadName(), timer);
```

Comment:
Shall we put this in try-catch block? Because these two statements can throw exceptions.